From 552a2f5f8b136263be1516995296adf4ee39b142 Mon Sep 17 00:00:00 2001
From: Gavin Hurlbut <ghurlbut@mythtv.org>
Date: Sat, 22 Jan 2011 00:04:10 -0800
Subject: [PATCH] Fix the use of datadirect with no named input file

---
 mythtv/libs/libmythtv/datadirect.cpp |    8 +++++---
 1 files changed, 5 insertions(+), 3 deletions(-)

diff --git a/mythtv/libs/libmythtv/datadirect.cpp b/mythtv/libs/libmythtv/datadirect.cpp
index 7257606..1e5954b 100644
--- a/mythtv/libs/libmythtv/datadirect.cpp
+++ b/mythtv/libs/libmythtv/datadirect.cpp
@@ -987,7 +987,9 @@ bool DataDirectProcessor::DDPost(
     password.replace('\'', "'\\''");
     userid.replace('\'', "'\\''");
 
-    QString outfile = (inputFile.isEmpty() ? "/tmp/crap" : inputFile);
+    if (inputFile.isEmpty()) {
+        inputFile = QString("/tmp/mythtv_ddp_data");
+    }
 
     QString command;
     {
@@ -996,7 +998,7 @@ bool DataDirectProcessor::DDPost(
             "wget --http-user='%1' --http-passwd='%2' --post-file='%3' "
             " %4 --user-agent='%5' --output-document=%6")
             .arg(userid).arg(password).arg(postFilename).arg(ddurl)
-            .arg(user_agent).arg(outfile);
+            .arg(user_agent).arg(inputFile);
     }
 
 #ifdef USING_MINGW
@@ -1022,7 +1024,7 @@ bool DataDirectProcessor::DDPost(
         return false;
 
 #ifndef USING_MINGW
-    command = QString("gzip -d %1").arg(outfile+".gz");
+    command = QString("gzip -df %1").arg(inputFile+".gz");
     if (myth_system(command) != GENERIC_EXIT_OK)
         return false;
 #endif
-- 
1.7.0.4

