Index: configure
===================================================================
--- configure	(revision 25841)
+++ configure	(working copy)
@@ -4745,7 +4745,7 @@
 cat >> external/FFmpeg/config.mak <<EOF
 LIBDIR=\$(INSTALL_ROOT)$libdir
 SHLIBDIR=\$(INSTALL_ROOT)$shlibdir
-INCDIR=\$(INSTALL_ROOT)$incdir
+INCDIR=$incdir
-BINDIR=\$(INSTALL_ROOT)$bindir
+BINDIR=$bindir
-DATADIR=\$(INSTALL_ROOT)$datadir
+DATADIR=$datadir
 MANDIR=\$(INSTALL_ROOT)$mandir

