Index: libs/libmythtv/programdata.cpp
===================================================================
--- libs/libmythtv/programdata.cpp	(revision 26134)
+++ libs/libmythtv/programdata.cpp	(working copy)
@@ -910,7 +910,7 @@
     QDateTime newTo   = to.addSecs(secs);
 
     MSqlQuery query(MSqlQuery::InitCon());
-    query.prepare("DELETE FROM program "
+    query.prepare("DELETE QUICK FROM program "
                   "WHERE starttime >= :FROM AND starttime < :TO "
                   "AND chanid = :CHANID ;");
     query.bindValue(":FROM", newFrom);
@@ -918,7 +918,7 @@
     query.bindValue(":CHANID", chanid);
     bool ok = query.exec();
 
-    query.prepare("DELETE FROM programrating "
+    query.prepare("DELETE QUICK FROM programrating "
                   "WHERE starttime >= :FROM AND starttime < :TO "
                   "AND chanid = :CHANID ;");
     query.bindValue(":FROM", newFrom);
@@ -926,7 +926,7 @@
     query.bindValue(":CHANID", chanid);
     ok &= query.exec();
 
-    query.prepare("DELETE FROM credits "
+    query.prepare("DELETE QUICK FROM credits "
                   "WHERE starttime >= :FROM AND starttime < :TO "
                   "AND chanid = :CHANID ;");
     query.bindValue(":FROM", newFrom);
@@ -934,7 +934,7 @@
     query.bindValue(":CHANID", chanid);
     ok &= query.exec();
 
-    query.prepare("DELETE FROM programgenres "
+    query.prepare("DELETE QUICK FROM programgenres "
                   "WHERE starttime >= :FROM AND starttime < :TO "
                   "AND chanid = :CHANID ;");
     query.bindValue(":FROM", newFrom);
@@ -949,12 +949,45 @@
     uint sourceid, const QDateTime &from, const QDateTime &to,
     bool use_channel_time_offset)
 {
-    vector<uint> chanids = ChannelUtil::GetChanIDs(sourceid);
-
+    MSqlQuery query(MSqlQuery::InitCon());
+    QString prepare, prepare2;
+    QList<QString> tables;
+    tables << "program" << "programrating" << "credits" << "programgenres";
     bool ok = true;
-    for (uint i = 0; i < chanids.size(); i++)
-        ok &= ClearDataByChannel(chanids[i], from, to, use_channel_time_offset);
 
+    prepare ="DELETE QUICK FROM %1 "
+             "USING %1 INNER JOIN channel "
+             "WHERE channel.sourceid = :SOURCEID "
+             "AND channel.xmltvid != '' "
+             "AND channel.tmoffset = 0 "
+             "AND %1.chanid = channel.chanid "
+             "AND %1.starttime >= :FROM "
+             "AND %1.starttime < :TO";
+
+    prepare2 ="DELETE QUICK FROM %1 "
+              "USING %1 INNER JOIN channel "
+              "WHERE channel.sourceid = :SOURCEID "
+              "AND channel.xmltvid != '' "
+              "AND channel.tmoffset != 0 "
+              "AND %1.chanid = channel.chanid "
+              "AND %1.starttime >= "
+              "FROM_UNIXTIME(UNIX_TIMESTAMP(:FROM)+channel.tmoffset) "
+              "AND %1.starttime < "
+              "FROM_UNIXTIME(UNIX_TIMESTAMP(:TO)+channel.tmoffset) ";
+
+    for (int i = 0; i < tables.size(); ++i) {
+        query.prepare(prepare.arg(tables.at(i)));
+        query.bindValue(":FROM", from);
+        query.bindValue(":TO", to);
+        query.bindValue(":SOURCEID", sourceid);
+        ok &= query.exec();
+
+        query.prepare(prepare2.arg(tables.at(i)));
+        query.bindValue(":FROM", from);
+        query.bindValue(":TO", to);
+        query.bindValue(":SOURCEID", sourceid);
+        ok &= query.exec();
+    }
     return ok;
 }
 
