commit 6a01060023ed451031cb5440448e8ed8dedd68aa
Author: Mario Limonciello <superm1@ubuntu.com>
Date:   Mon Feb 13 13:56:24 2012 -0600

    update ABI

diff --git a/mythtv/settings.pro b/mythtv/settings.pro
index 492aa94..8b80046 100644
--- a/mythtv/settings.pro
+++ b/mythtv/settings.pro
@@ -38,8 +38,8 @@ isEmpty( LIBDIR ) {
     LIBDIR = $${RUNPREFIX}/$${LIBDIRNAME}
 }
 
-LIBVERSION = 0.24
-VERSION = 0.24.0
+LIBVERSION = 0.25
+VERSION = 0.25.0
 
 isEmpty(TARGET_OS) : win32 {
     CONFIG += mingw
