Index: mythtv-trunk/configure
===================================================================
--- mythtv-trunk.orig/configure	2010-08-25 09:57:29.718118588 -0500
+++ mythtv-trunk/configure	2010-08-25 10:26:41.470129064 -0500
@@ -4787,6 +4787,7 @@
   mkdir config
 fi
 cat - > config/Makefile <<END
+check:
 
 all:
 qmake_all:
Index: mythtv-trunk/external/Makefile
===================================================================
--- mythtv-trunk.orig/external/Makefile	2010-08-25 10:29:23.930118883 -0500
+++ mythtv-trunk/external/Makefile	2010-08-25 10:29:30.577148874 -0500
@@ -32,4 +32,6 @@
 $(addsuffix _uninstall,$(SUBDIRS)):
 	$(MAKE) -C $(subst _uninstall,,$@) uninstall
 
-uninstall: $(addsuffix _uninstall,$(SUBDIRS))
\ No newline at end of file
+uninstall: $(addsuffix _uninstall,$(SUBDIRS))
+
+check:
