Index: mythtv-master/mythplugins/mythweb/mythweb.conf.apache
===================================================================
--- mythtv-master.orig/mythplugins/mythweb/mythweb.conf.apache	2012-02-14 15:12:29.242678338 -0600
+++ mythtv-master/mythplugins/mythweb/mythweb.conf.apache	2012-02-14 15:13:24.266950841 -0600
@@ -26,10 +26,10 @@
 #    /var/www/html/mythweb
 #    /srv/www/htdocs/mythweb
 #
-    <Directory "/var/www/html/data">
+    <Directory "/var/www/mythweb/data">
         Options -All +FollowSymLinks +IncludesNoExec
     </Directory>
-    <Directory "/var/www/html" >
+    <Directory "/var/www/mythweb" >
 
     ############################################################################
     # I *strongly* urge you to turn on authentication for MythWeb.  It is disabled
