Patch to include layout option for python 2.6
==========================================================================
Index: mythtv-trunk/mythtv/bindings/python/Makefile
===================================================================
--- mythtv-trunk.orig/mythtv/bindings/python/Makefile	2010-12-30 02:46:20.388115044 -0600
+++ mythtv-trunk/mythtv/bindings/python/Makefile	2010-12-30 02:46:48.900115044 -0600
@@ -1,7 +1,7 @@
 include ../../config.mak
 
 ifdef INSTALL_ROOT
-	ROOT_FLAGS = --root="$(INSTALL_ROOT)"
+	ROOT_FLAGS = --root="$(INSTALL_ROOT)/../libmyth-python"
 endif
 
 REAL_PREFIX = $(PREFIX)
@@ -10,6 +10,7 @@
 		PREFIX_FLAGS=--prefix="$(PREFIX)"
 	endif
 endif
+PREFIX_FLAGS="--install-layout=deb"
 
 all: python_build
 
Index: mythtv-trunk/mythtv/bindings/perl/Makefile
===================================================================
--- mythtv-trunk.orig/mythtv/bindings/perl/Makefile	2010-12-30 02:46:20.428115044 -0600
+++ mythtv-trunk/mythtv/bindings/perl/Makefile	2010-12-30 02:46:48.900115044 -0600
@@ -1,7 +1,7 @@
 include ../../config.mak
 
 ifdef INSTALL_ROOT
-ROOT_FLAGS = PERL_INSTALL_ROOT="$(INSTALL_ROOT)"
+ROOT_FLAGS = PERL_INSTALL_ROOT="$(INSTALL_ROOT)/../libmythtv-perl"
 endif
 
 ifneq ($(PREFIX:/=), /usr)
Index: mythtv-trunk/mythtv/bindings/php/Makefile
===================================================================
--- mythtv-trunk.orig/mythtv/bindings/php/Makefile	2010-12-30 02:46:55.096115045 -0600
+++ mythtv-trunk/mythtv/bindings/php/Makefile	2010-12-30 02:47:13.364115044 -0600
@@ -4,6 +4,7 @@
 FILES += MythTV.php MythTVProgram.php MythTVRecording.php
 FILES += MythTVStorageGroup.php
 
+INSTALL_ROOT = ${INSTALL_ROOT}../php-mythtv
 INSTDIR = ${INSTALL_ROOT}${PREFIX}/share/mythtv/bindings/php
 INSTALL = /usr/bin/install
 
