Patch to include layout option for python 2.6
==========================================================================
Index: mythtv-trunk/bindings/python/Makefile
===================================================================
--- mythtv-trunk.orig/bindings/python/Makefile	2010-10-10 17:34:11.000000000 -0500
+++ mythtv-trunk/bindings/python/Makefile	2010-10-14 19:31:41.622273000 -0500
@@ -1,7 +1,7 @@
 include ../../config.mak
 
 ifdef INSTALL_ROOT
-	ROOT_FLAGS = --root="$(INSTALL_ROOT)"
+	ROOT_FLAGS = --root="$(INSTALL_ROOT)/../libmyth-python"
 endif
 
 REAL_PREFIX = $(PREFIX)
@@ -10,6 +10,7 @@
 		PREFIX_FLAGS=--prefix="$(PREFIX)"
 	endif
 endif
+PREFIX_FLAGS="--install-layout=deb"
 
 all: python_build
 
Index: mythtv-trunk/bindings/perl/Makefile
===================================================================
--- mythtv-trunk.orig/bindings/perl/Makefile	2010-09-22 17:03:20.000000000 -0500
+++ mythtv-trunk/bindings/perl/Makefile	2010-10-14 19:32:12.650273001 -0500
@@ -1,7 +1,7 @@
 include ../../config.mak
 
 ifdef INSTALL_ROOT
-ROOT_FLAGS = PERL_INSTALL_ROOT="$(INSTALL_ROOT)"
+ROOT_FLAGS = PERL_INSTALL_ROOT="$(INSTALL_ROOT)/../libmythtv-perl"
 endif
 
 ifneq ($(PREFIX:/=), /usr)
