Index: mythtv-trunk/libs/libmyth/mythdeque.h
===================================================================
--- mythtv-trunk.orig/libs/libmyth/mythdeque.h	2009-04-10 19:04:30.000000000 -0500
+++ mythtv-trunk/libs/libmyth/mythdeque.h	2010-08-25 09:56:11.233121070 -0500
@@ -19,7 +19,7 @@
     T dequeue()
     {
         if (deque<T>::empty())
-            return (T)(0);
+            return T();
         T item = deque<T>::front();
         deque<T>::pop_front();
         return item;
Index: mythtv-trunk/libs/libmythtv/recordingprofile.h
===================================================================
--- mythtv-trunk.orig/libs/libmythtv/recordingprofile.h	2010-05-16 09:23:11.000000000 -0500
+++ mythtv-trunk/libs/libmythtv/recordingprofile.h	2010-08-25 09:56:11.233121070 -0500
@@ -84,7 +84,7 @@
 
   public:
     // initializers
-    RecordingProfile(QString profName = NULL);
+    RecordingProfile(QString profName = QString());
     virtual void loadByID(int id);
     virtual bool loadByType(const QString &name, const QString &cardtype);
     virtual bool loadByGroup(const QString &name, const QString &group);
Index: mythtv-trunk/libs/libmythtv/tv_rec.h
===================================================================
--- mythtv-trunk.orig/libs/libmythtv/tv_rec.h	2010-08-23 16:11:09.000000000 -0500
+++ mythtv-trunk/libs/libmythtv/tv_rec.h	2010-08-25 09:56:11.233121070 -0500
@@ -93,7 +93,7 @@
 class TuningRequest
 {
   public:
-    TuningRequest(uint f) :
+    TuningRequest(uint f = 0) :
         flags(f), program(NULL), channel(QString::null),
         input(QString::null), majorChan(0), minorChan(0), progNum(-1) {;}
     TuningRequest(uint f, RecordingInfo *p) :
Index: mythtv-trunk/libs/libmythupnp/eventing.h
===================================================================
--- mythtv-trunk.orig/libs/libmythupnp/eventing.h	2009-12-30 23:38:50.000000000 -0600
+++ mythtv-trunk/libs/libmythupnp/eventing.h	2010-08-25 09:56:11.233121070 -0500
@@ -233,7 +233,7 @@
         {
             SVMap::iterator it = m_map.find(sName);
             if (it == m_map.end())
-                return T(0);
+                return T();
 
             StateVariable< T > *pVariable =
                 dynamic_cast< StateVariable< T > *>( *it );
@@ -241,7 +241,7 @@
             if (pVariable != NULL)
                 return pVariable->GetValue();
 
-            return T(0);
+            return T();
         }
 
         uint BuildNotifyBody(QTextStream &ts, TaskTime ttLastNotified) const;
Index: mythtv-trunk/libs/libmyth/audiooutputbase.cpp
===================================================================
--- mythtv-trunk.orig/libs/libmyth/audiooutputbase.cpp	2010-08-25 09:56:53.405135316 -0500
+++ mythtv-trunk/libs/libmyth/audiooutputbase.cpp	2010-08-25 09:57:08.217149671 -0500
@@ -75,7 +75,7 @@
     source_bytes_per_frame(0),
     needs_upmix(false),         needs_downmix(false),
     surround_mode(QUALITY_LOW), old_stretchfactor(1.0f),
-    volume(80),                 volumeControl(NULL),
+    volume(80),                 volumeControl(QString()),
 
     processing(false),
 
