Index: mythtv-fixes/libs/libmyth/mythdeque.h
===================================================================
--- mythtv-fixes.orig/libs/libmyth/mythdeque.h	2010-08-24 19:12:55.017128343 -0500
+++ mythtv-fixes/libs/libmyth/mythdeque.h	2010-08-24 19:13:17.590120714 -0500
@@ -19,7 +19,7 @@
     T dequeue()
     {
         if (deque<T>::empty())
-            return (T)(0);
+            return T();
         T item = deque<T>::front();
         deque<T>::pop_front();
         return item;
Index: mythtv-fixes/libs/libmythtv/recordingprofile.h
===================================================================
--- mythtv-fixes.orig/libs/libmythtv/recordingprofile.h	2010-08-24 19:13:05.190144174 -0500
+++ mythtv-fixes/libs/libmythtv/recordingprofile.h	2010-08-24 19:13:17.610149724 -0500
@@ -84,7 +84,7 @@
 
   public:
     // initializers
-    RecordingProfile(QString profName = NULL);
+    RecordingProfile(QString profName = QString());
     virtual void loadByID(int id);
     virtual bool loadByType(const QString &name, const QString &cardtype);
     virtual bool loadByGroup(const QString &name, const QString &group);
Index: mythtv-fixes/libs/libmythtv/tv_rec.h
===================================================================
--- mythtv-fixes.orig/libs/libmythtv/tv_rec.h	2010-08-24 19:13:10.173137963 -0500
+++ mythtv-fixes/libs/libmythtv/tv_rec.h	2010-08-24 19:13:17.622149153 -0500
@@ -106,7 +106,7 @@
 class TuningRequest
 {
   public:
-    TuningRequest(uint f) :
+    TuningRequest(uint f = 0) :
         flags(f), program(NULL), channel(QString::null),
         input(QString::null), majorChan(0), minorChan(0), progNum(-1) {;}
     TuningRequest(uint f, RecordingInfo *p) :
Index: mythtv-fixes/libs/libmythupnp/eventing.h
===================================================================
--- mythtv-fixes.orig/libs/libmythupnp/eventing.h	2010-08-24 19:12:41.501129379 -0500
+++ mythtv-fixes/libs/libmythupnp/eventing.h	2010-08-24 19:13:17.586121044 -0500
@@ -233,7 +233,7 @@
         {
             SVMap::iterator it = m_map.find(sName);
             if (it == m_map.end())
-                return T(0);
+                return T();
 
             StateVariable< T > *pVariable =
                 dynamic_cast< StateVariable< T > *>( *it );
@@ -241,7 +241,7 @@
             if (pVariable != NULL)
                 return pVariable->GetValue();
 
-            return T(0);
+            return T();
         }
 
         uint BuildNotifyBody(QTextStream &ts, TaskTime ttLastNotified) const;
