Index: mythtv/mythtv/libs/libmythui/mythuihelper.h
===================================================================
--- mythtv.orig/mythtv/libs/libmythui/mythuihelper.h	2015-04-12 15:24:03.598909276 +0000
+++ mythtv/mythtv/libs/libmythui/mythuihelper.h	2015-04-12 15:24:03.598909276 +0000
@@ -9,8 +9,8 @@
 #include "mythuiexp.h"
 #include "themeinfo.h"
 
-#define DEFAULT_UI_THEME "Terra"
-#define FALLBACK_UI_THEME "MythCenter-wide"
+#define DEFAULT_UI_THEME "Mythbuntu"
+#define FALLBACK_UI_THEME "Terra"
 
 class MythUIHelperPrivate;
 class MThreadPool;
