Filed upstream here:
https://code.mythtv.org/trac/ticket/12696

Index: mythtv/mythplugins/configure
===================================================================
--- mythtv.orig/mythplugins/configure
+++ mythtv/mythplugins/configure
@@ -722,7 +722,7 @@ if enabled music ; then
 fi
 
 if enabled zoneminder ; then
-    if ! check_header mysql/mysql.h ; then
+    if ! check_header mysql/mysql.h -I/usr/include/mysql; then
         echo "Disabling MythZoneMinder. Requires mysql/mysql.h header"
         disable zoneminder
     fi
Index: mythtv/mythplugins/mythzoneminder/mythzmserver/mythzmserver.pro
===================================================================
--- mythtv.orig/mythplugins/mythzoneminder/mythzmserver/mythzmserver.pro
+++ mythtv/mythplugins/mythzoneminder/mythzmserver/mythzmserver.pro
@@ -10,6 +10,8 @@ target.path = $${PREFIX}/bin
 
 INSTALLS = target
 
+INCLUDEPATH += /usr/include/mysql
+
 macx {
     CONFIG += qt
     QT += sql
Index: mythtv/mythplugins/mythzoneminder/mythzoneminder/mythzoneminder.pro
===================================================================
--- mythtv.orig/mythplugins/mythzoneminder/mythzoneminder/mythzoneminder.pro
+++ mythtv/mythplugins/mythzoneminder/mythzoneminder/mythzoneminder.pro
@@ -11,6 +11,7 @@ target.path = $${LIBDIR}/mythtv/plugins
 INSTALLS += target
 
 INCLUDEPATH += $${PREFIX}/include/mythtv/libmythui
+INCLUDEPATH += /usr/include/mysql
 
 # Input
 HEADERS += zmconsole.h zmplayer.h zmevents.h zmliveplayer.h zmdefines.h
